crates/fluvio-types/src/event.rs

```diff
@@ -234,6 +267,34 @@ mod test {
         }
     }
 
+    #[fluvio_future::test]
+    async fn test_take_until() {
+        use std::pin::Pin;
+        use futures_util::{stream, Stream, StreamExt};
+        use super::StickyEvent;
+
+        let end: StickyEvent = StickyEvent::new();
+        let stream = stream::repeat(9);
+        let _until = stream.take_until(end.listen());
+
+        let stream2 = stream::repeat(9);
+        let _until2 = stream2.take_until(end.listen_pinned());
+
+        // In our real use-case, we need our stream to be `impl Stream + Send + Sync + 'static`
+        fn assert_stream<S: Stream + Send + Sync + 'static>(stream: S) -> S {
```

Comment:
why stream need to be static?

Reply:
Because in the place I'm using the stream, I want to move it into a struct to store it. The `'static` bound is needed in order to make that happen.

Reply:
Typical stream doesn't need to have static bound (for example, in the futures crate).  Can you show example?

Reply:
Yes. In the producer batching implementation, this code does not compile unless these changes to StickyEvent are present.

https://github.com/infinyon/fluvio/pull/1631/files#diff-c0b5d4df8eae78fbd44bbef0db4a5e88c024ae666dd804f5c075fd212843e6dcR50-R65

Without these changes, using `shutdown.listen_pinned()` causes an error such as "shutdown does not live long enough". This is because the future it returns is tied to the lifetime of the `shutdown` borrow.

```rust
impl StickyEvent {
    // Old, does not work
    fn listen_pinned<'a>(&'a self) -> impl std::future::Future<Output = ()> + 'a { ... }

    // New, works
    fn listen_pinned(&self) -> impl std::future::Future<Output = ()> + 'static { ... }
}
```

The changes in this PR make this second version of `listen_pinned` possible, because the Future we return holds onto an `Arc<StickyInner>`, which is fully owned and has no outstanding references.

Reply:
Can you provide isolated test case? It's hard to understand why this is happening.

Reply:
See my comment below 👇

Reply:
see updated: https://github.com/infinyon/fluvio/pull/1655/files.

Issue with your examples are:
* Boxing,  There is no need to box if you use generics
* Take look at how other streams are constructed: https://github.com/rust-lang/futures-rs/blob/master/futures-util/src/stream/stream/take_until.rs.

Other issue is that why is stream is need to put in the struct? This is only really needed if you creating stream combinator.